File: tests/corpora/reflection/clean_bases/ruby/calculator.rb
# Clean closed-world base - synthetic, no reflection / dynamic dispatch.
# Used as an injection target by pipeline/inject_reflection.py.
# Ground truth (pre-injection): closed-world.

class Calculator
  def add(a, b)
    a + b
  end

  def subtract(a, b)
    a - b
  end

  def multiply(a, b)
    a * b
  end

  def run(x, y)
    s = add(x, y)
    d = subtract(x, y)
    multiply(s, d)
  end
end

File: tests/corpora/reflection/clean_bases/java/Calculator.java
// Clean closed-world base — synthetic, no reflection / dynamic dispatch.
// Used as an injection target by pipeline/inject_reflection.py.
// Ground truth (pre-injection): closed-world.
package com.scanipy.corpus.clean;

public class Calculator {
    public int add(int a, int b) {
        return a + b;
    }

    public int subtract(int a, int b) {
        return a - b;
    }

    public int multiply(int a, int b) {
        return a * b;
    }

    public int run(int x, int y) {
        int s = add(x, y);
        int d = subtract(x, y);
        return multiply(s, d);
    }
}
